javaparser-core-generators/src/main/java/com/github/javaparser/generator/core/visitor/HashCodeVisitorGenerator.java

```diff
@@ -22,19 +22,29 @@ public HashCodeVisitorGenerator(SourceRoot sourceRoot) {
     }
 
     @Override
-    protected void generateVisitMethodBody(BaseNodeMetaModel node, MethodDeclaration visitMethod, CompilationUnit compilationUnit) {
+    protected void generateVisitMethodBody(BaseNodeMetaModel node, MethodDeclaration visitMethod,
+                                           CompilationUnit compilationUnit) {
         visitMethod.getParameters().forEach(p -> p.setFinal(true));
 
         final BlockStmt body = visitMethod.getBody().get();
         body.getStatements().clear();
 
         final SeparatedItemStringBuilder builder = new SeparatedItemStringBuilder("return ", "* 31 +", ";");
-        final List<PropertyMetaModel> propertyMetaModels= node.getAllPropertyMetaModels();
-        if (propertyMetaModels.isEmpty()) {
+        final List<PropertyMetaModel> propertyMetaModels = node.getAllPropertyMetaModels();
+        String ndName = node.getTypeName();
+        if (ndName.equals("JavadocComment") || ndName.equals("LineComment") || ndName.equals("BlockComment")
+                || propertyMetaModels.isEmpty()) {
             builder.append("0");
         } else {
             for (PropertyMetaModel field : propertyMetaModels) {
                 final String getter = field.getGetterMethodName() + "()";
+                if (getter.equals("getComment()")) {
```

Comment:
Same thing here: you should be able to find the `PropertyMetaModel` for `comment` in `Node`.

Reply:
@matozoid: I haven't tested it, but something like this:

```
if (field.equals(JavaParserMetaModel.nodeMetaModel.commentPropertyMetaModel)) {
}
```